### DISCUSSION TITLE
Javascript in a form or Screen

### DISCUSSION THREAD

**Discussion by PGLange** [Budibase/budibase#1795] [Ideas]
Hey there,
I'm making progress in using Budibase, got tables and Screens and Automatisations. I was asked to layout the forms, that number-fields should show numbers with a thounder Seperator (1.000.000) and I should prevent inputs like 0,23 (to ensure whole numbers). 
How can I archive this? 
I read the budiboard documentation, but ... I didn't find anything on this, but sometime I tend to overlook stuff. 
The only Idea is, to wrap Budibase in a page that contains these Scripts - like using pico cms - but in the end this is external code manipulating fileds - Can't stand this idea :)
May you help me out?

**Comment by mjashanks** [comment]
Hey @PGLange 

The things that you are asking for are not currently support for Inputs. 
- Inputs do not accept a thousands separator `1,000,000`
- Inputs cannot display a thousands separator, when data is loaded in
- Inputs cannot define a "Number of Decimal places"

These are very sensible suggestions, I think that we should add this.

If you want to display a number with a thousand separator, in a non-input, e.g. a `Paragraph`, then you can do so with a binding expression:
- `{{ toPrecision Form.Fields.Amount 0 }}` will display the `Form.Fields.Amount` field with 0 decimal places
- `{{ addCommas Form.Fields.Amount }}` will add a thousands separator - `1,000,000`

**Reply by mjashanks** [reply]
...also related to https://github.com/Budibase/budibase/discussions/1679

**Reply by PGLange** [reply]
Solution to  thousands separator with a dott (".") is
{{replace (addCommas Some.Field) "," "."}}

**Comment by phamalda** [comment]
G'morning, BudiMates. Hope you are all doing well and staying warm. I wanted to revisit this topic. I am trying to get commas appropriately placed in dollar figures in a value coming from the internal database. I can't seem to get the logic down. I have tried the syntax below, but both break my logic, and my output simply changes to the logic from the code. 

Revenue: ${{ addCommas [Sigmatech Revenue]}}

Revenue: ${{ replace (addCommas [Sigmatech Revenue]) "," "."}}

If I use this:

Revenue: ${{ addCommas 1000000 ]}}

this works just fine. I'm sure that I am missing something really simple. Any help would be greatly appreciated. Thanks in advance.

**Reply by mikesealey** [reply]
Hey @phamalda - try looking into `toLocaleString()` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

**Reply by phamalda** [reply]
Thanks for the quick response, @mikesealey. I've learned that I can use the syntax below...as long as that attribute actually contains a value. I just need to find out how to handle NULL values.

Revenue: ${{  addCommas  [Sigmatech Revenue] }}

Thanks again. 👍
